velox/exec/StreamingAggregation.cpp

```diff
@@ -341,15 +345,20 @@ RowVectorPtr StreamingAggregation::getOutput() {
   masks_->addInput(input_, inputRows_);
 
   const auto numPrevGroups = numGroups_;
-
   assignGroups();
   initializeNewGroups(numPrevGroups);
   evaluateAggregates();
 
   RowVectorPtr output;
-  if (numGroups_ > minOutputBatchSize_) {
-    output = createOutput(
-        std::min(numGroups_ - 1, static_cast<size_t>(maxOutputBatchSize_)));
+  if (trySplitOutputAtInputBoundary_) {
+    if ((numPrevGroups != 0) && (numGroups_ > minOutputBatchSize_)) {
+      output = createOutput(std::min(numGroups_ - 1, numPrevGroups));
+    }
+  } else {
+    if (numGroups_ > minOutputBatchSize_) {
+      output = createOutput(
+          std::min(numGroups_ - 1, static_cast<size_t>(maxOutputBatchSize_)));
+    }
```

Comment:
I think we can extract this section out as a separate batching method that deals with different batching strategies based on settings.